### ISSUE TITLE
Define a style to resize only if image is larger than style size

### ISSUE THREAD

**Issue by flyingL123** [czim/laravel-paperclip#14]
I'm wondering if this library supports my use case. I would like to resize images proportionally down to a width of 1280px only if the original image is larger than 1280px.

As you can see I never found a built-in way to do this using stapler. I had to use a custom closure to get it working.

https://github.com/CodeSleeve/stapler/issues/161

Does laravel-paperclip support this or will I have to use the same custom solution. It would be very nice to simply define the dimensions as `1280>` and not have to use a closure.

Any idea?

**Comment by czim** [comment]
Yes and no.

Yes, because paperclip supports pretty much *any* modification that you want to make -- you can make your own variant strategies. Just add a simple class (extend the [ImageResizeStrategy](https://github.com/czim/file-handling/blob/master/src/Variant/Strategies/ImageResizeStrategy.php), for instance and configure an alias for it under the `variants` key in the [paperclip config](https://github.com/czim/laravel-paperclip/blob/master/config/paperclip.php).

No, because this use-case does not come with the package -- you will have to program it yourself. If you need help getting it set up with the configuration file, just ask. I just noticed that I didn't write up an example in the documentation, though the code and configuration are pretty clear on where the logic can be 'injected'.

**Comment by flyingL123** [comment]
Thanks a lot, this is very helpful. I didn't fully grasp the Strategy concept and how those map to aliases, but I think I do now from your explanation. An explanation in the docs would be great.

I'll give this a shot and let you know if any issues arise.

**Comment by czim** [comment]
Yes, that is correct. The interface is uniform, and demands an options array. So that is precisely the reason, as you say.

**Comment by flyingL123** [comment]
Hi @czim, I am trying to implement this now using a new custom Variant. What is the best way to get the current size of the file from with the variant? I'm currently using `getimagesize`, but I'm not sure if this takes into account any previous changes that have been made to the file by other variants. For example, if the image is first autorated, then passed to my variant, will this method get the correct width, or should I be getting the width on some in-memory version of the file that is being processed?

```
<?php

namespace App;

use Czim\FileHandling\Variant\Strategies\ImageResizeStrategy;

class MinimumWidthImageResizeStrategy extends ImageResizeStrategy
{
    protected function perform()
    {
        list($width, $height) = getimagesize($this->file->path());

        if ($width > 1280) {
            return parent::perform();
        }

        return true;
    }
}
```
